src/main/java/org/spongepowered/common/registry/RegistryMapImpl.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.registry;
+
+import org.spongepowered.api.CatalogKey;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+final class RegistryMapImpl<V> extends HashMap<CatalogKey, V> implements RegistryMap<V> {
```

Comment:
Why do we have an interface with exactly one implementation which is essentially the trivial implementation. For that matter why do we have `RegistryMap` in the first place, it seems to just be a `Map<CatalogKey, V>` with the only additional method being `forCatalogRegistration` which just converts it to a map keyed on strings. Which could be concerted to a utility method, or even better since it is used in exactly one place just inlined into `provideCatalogMap`

Reply:
The `RegistryMap` interface and `RegistryMapImpl` came from my old PR. In any case, it handles things like ensuring `values` is unmodifiable, `getOptional` wrapper, and `forCatalogRegistration` currently. It results in a lot less duplicated code throughout all the modules.
It can be collapsed into one class now that it isn't a `String` key.